### ISSUE TITLE
Fragment Doc get lost after upgrading above the 1.9.1

### ISSUE THREAD

**Issue by ivansky** [dotansimha/graphql-code-generator#3259]
**Describe the bug**
On 1.9.1 there is an import from Graphql file to use `SomeFragmentDoc` string.
After upgrading above 1.9.1 it doesn't work properly, I didn't tested alpha version, there are too much of them.
I don't know maybe it's intended behaviour and I need to use some kinda import feature, but I didn't need before.

**To Reproduce**
There is a repo
https://github.com/ivansky/graphql-codegen-test-repo/tree/babc5efd6e757bcfb1fe3dcfeac2851f00f6866b

And commit that describes the issue
https://github.com/ivansky/graphql-codegen-test-repo/commit/babc5efd6e757bcfb1fe3dcfeac2851f00f6866b

1. My GraphQL schema:

```graphql
schema {
    query: Query
}

type Query {
    check(
        product: ProductCheckInput!
    ): ProductCheckResult!
}

interface Service {
    id: ID!
    productId: String!
}

type AService implements Service {
    id: ID!
    productId: String!
    dooms: [[ADoom!]!]!
}

type ALoad {
    code: String!
}

type ADoom {
    loads: [ALoad!]!
    code: String!
    description: String
    searchCount: Int
}

type Product {
    rules: [String!]!
    id: ID!
    notes: [String!]!
    services: [Service!]!
}

type ProductCheckResult {
    product: Product
    success: Boolean!
}

input ProductCheckInput {
    searchId: String!
    searchProductId: String!
}

scalar Date

scalar DateTime

scalar Time

type Auth {
    accessToken: String!
}

type Client {
    auth: Auth
}

extend type Query {
    client: Client!
}

```

2. My GraphQL operations:

**AServiceFragment.graphql**
```graphql
fragment AServiceFragment on AService {
    id
    productId
}
```

**CheckProduct.graphql**
```graphql
query CheckProduct($product: ProductCheckInput!) {
    check(product: $product) {
        success
        product {
            notes
            rules
            services {
                ...AServiceFragment
                ...on AService {
                    dooms {
                        code
                        loads {
                            code
                        }
                    }
                }
            }
        }
    }
}
```

3. My `codegen.yml` config file:

```yml
overwrite: true
config:
  namingConvention:
    typeNames: change-case#pascalCase
    transformUnderscore: true
schema:
  - ./schema.api.graphql
  - ./schema.client.graphql
generates:
  types/types.ts:
    documents:
      - './graphql/operations/**/*.graphql'
    schema:
      - ./schema.api.graphql
      - ./schema.client.graphql
    plugins:
      - typescript
  types/:
    documents:
      - './graphql/operations/**/*.graphql'
    schema:
      - ./schema.api.graphql
      - ./schema.client.graphql
    config:
      noNamespaces: true
      withHooks: true
      withComponent: true
      withHOC: false
    preset: near-operation-file
    presetConfig:
      extension: .generated.tsx
      baseTypesPath: types.ts
    plugins:
      - typescript-operations
      - typescript-react-apollo
  types/resolversTypes.ts:
    config:
      contextType: ./graphql/contexts/ClientContext#ClientContext
    schema:
      - ./schema.client.graphql
    plugins:
      - typescript
      - typescript-resolvers
  graphql/introspectionResult.generated.ts:
    schema:
      - ./schema.api.graphql
    plugins:
      - fragment-matcher
```

**Expected behavior**
Everything that is used within the module should be imported automatically.

**Environment:**

- OS: MacOS
- `@graphql-codegen/...`: Any above 1.9.1 
- NodeJS: 13.5.0

**Comment by xiaoyu-tamu** [comment]
I can confirm version 1.11.2 have the same issue.

**Comment by dotansimha** [comment]
@ardatan can you please take a look?

**Comment by Eddman** [comment]
The problem still persists in 1.12.1.

**Comment by Dema** [comment]
I did a bit of digging, for me it looks like a bit of incomplete refactor https://github.com/dotansimha/graphql-code-generator/issues/3426

**Comment by Dema** [comment]
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=86691508)](https://www.bountysource.com/issues/86691508-fragment-doc-get-lost-after-upgrading-above-the-1-9-1)

**Comment by ardatan** [comment]
Could you try the following canary version?
`1.12.2-alpha-f5f1494a.14`

**Comment by lkleuver** [comment]
this seems to work for me.
though this issue (https://github.com/dotansimha/graphql-code-generator/issues/3066) which seems related is not working yet.

**Comment by ardatan** [comment]
@lkleuver Let's keep your problem in that issue #3066 seperately. That PR only covers this issue for now.

**Comment by Dema** [comment]
@ardatan Yes, it works for me! Thank you very much!

**Comment by Dema** [comment]
Was a bit too early to celebrate :)
```
import { PublicPropertyFieldsDoc,PublicPropertyFieldsFragment } from '../../../../../../core/graphql/fragments/public-property-fields.generated';
``` 
this is the generated import.

These are exports
```
export type PublicPropertyFieldsFragment = (
  { readonly __typename?: 'property_api_PropertyEnrichedOutput' }
```
and 
```
export const PublicPropertyFieldsFragmentDoc = gql`
    fragment PublicPropertyFields on property_api_PropertyEnrichedOutput {
```

and it gives me an error
```
Module '"../../../../../../core/graphql/fragments/public-property-fields.generated"' has no exported member 'PublicPropertyFieldsDoc'.```

**Comment by ardatan** [comment]
My bad, I gave you the older version. Please try that one
`1.12.2-alpha-ea7264f9.15`

**Comment by Dema** [comment]
Yep! This version works! Thank you!!

**Comment by henry-young** [comment]
> My bad, I gave you the older version. Please try that one
> `1.12.2-alpha-ea7264f9.15`

@ardatan I can also confirm that `1.12.2-alpha-ea7264f9.15` fixed this issue! Thanks, look forward to the full release.

**Comment by dotansimha** [comment]
Fixed in `1.12.2`! Thanks @ardatan

**Comment by Dema** [comment]
@ardatan Could you please go and claim the bounty? :smile_cat:
